zha/application/platforms/sensor/__init__.py

```diff
@@ -502,16 +515,85 @@ def formatter(self, value: int) -> str | None:
         return self._enum(value).name
 
 
-@MULTI_MATCH(
-    cluster_handler_names=CLUSTER_HANDLER_ANALOG_INPUT,
-    manufacturers="Digi",
-    stop_on_match_group=CLUSTER_HANDLER_ANALOG_INPUT,
-)
-class AnalogInput(Sensor):
+@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_MULTISTATE_INPUT)
+class MultiStateInputSensor(EnumSensor):
+    """Sensor that displays enumerated values."""
+
+    _attribute_name = "present_value"
+    _unique_id_suffix = "present_value"
+    _attr_entity_registry_enabled_default = False
+    _attr_traslation_key = "multistate_input"
+    _enum = enum.Enum("Empty", names=())  # type: ignore [misc]
+
+    def __init__(
+        self,
+        unique_id: str,
+        cluster_handlers: list[ClusterHandler],
+        endpoint: Endpoint,
+        device: Device,
+        **kwargs: Any,
+    ) -> None:
+        """Init this sensor."""
+        super().__init__(unique_id, cluster_handlers, endpoint, device, **kwargs)
+        if self._cluster_handler.cluster.get("state_text"):
+            self._enum = enum.Enum(  # type: ignore [misc]
+                "state_text", self._cluster_handler.cluster.get("state_text")
+            )
+        elif self._cluster_handler.cluster.get("number_of_states") is not None:
+            self._enum = enum.Enum(  # type: ignore [misc]
+                "state_text",
+                [
+                    (f"state_{i+1}", i + 1)
+                    for i in range(
+                        self._cluster_handler.cluster.get("number_of_states")
+                    )
+                ],
+            )
+
+
+@CONFIG_DIAGNOSTIC_MATCH(cluster_handler_names=CLUSTER_HANDLER_ANALOG_INPUT)
+class AnalogInputSensor(Sensor):
     """Sensor that displays analog input values."""
 
     _attribute_name = "present_value"
     _attr_translation_key: str = "analog_input"
+    _unique_id_suffix = "present_value"
+    _attr_entity_registry_enabled_default = False
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        unique_id: str,
+        cluster_handlers: list[ClusterHandler],
+        endpoint: Endpoint,
+        device: Device,
+        **kwargs: Any,
+    ) -> None:
+        """Init this sensor."""
+        super().__init__(unique_id, cluster_handlers, endpoint, device, **kwargs)
+        engineering_units = self._cluster_handler.engineering_units
+        self._attr_native_unit_of_measurement = UNITS.get(engineering_units)
+        self._attr_state_class = SensorStateClass.MEASUREMENT
```

Comment:
Should this always be measurement? I can see a situation where total or total increasing makes sense, but I don't see a good way to choose based on the attributes we have available.